search-app/src/cmr/search/site/data.clj

```diff
@@ -25,8 +57,48 @@
 (defn cmr-link
   "Given a CMR host and a concept ID, return the collection landing page for
   the given id."
-  [cmr-host concept-id]
-  (format "https://%s/concepts/%s.html" cmr-host concept-id))
+  [cmr-base-url concept-id]
+  (format "%sconcepts/%s.html" cmr-base-url concept-id))
+
+(defn make-href
+  "Create the `href` part of a landing page link."
+  [cmr-base-url item]
+  (if (has-doi? item)
+    (doi-link (get-doi item))
+    (cmr-link cmr-base-url (get-in item [:meta :concept-id]))))
+
+(defn get-long-name
+  "Get a collection item's long name."
+  [item]
+  (or (get-in item [:umm "EntryTitle"])
```

Comment:
Fixed. Also renamed `long-name` instances to `entity-title`, per your suggestion in chat.